src/services/swap_service.ts

```diff
@@ -127,9 +134,10 @@ export class SwapService {
         let conservativeBestCaseGasEstimate = new BigNumber(worstCaseGas).plus(gasTokenGasCost);
         if (!skipValidation && from) {
             // Force a revert error if the takerAddress does not have enough ETH.
-            const txDataValue = isETHSell
-                ? BigNumber.min(value, await this._web3Wrapper.getBalanceInWeiAsync(from))
-                : value;
+            const txDataValue =
+                isETHSell || isETHBuy
+                    ? BigNumber.min(value, await this._web3Wrapper.getBalanceInWeiAsync(from))
+                    : value;
```

Comment:
why do we do this? and why does it matter if we're buying ETH?

Reply:
It doesn't matter if we're buying, so `isETHBuy` should be removed.

It once mattered when we were selling as geth allowed any `value` and was not validated. So a user wouldn't see any issue with selling 100 ETH (with no ETH balance) until hitting metamask.

Latest geth now validates `value` so this entire check can disappear soon.